onward/app/weather/geo/CitiesLookUpV2.scala

```diff
@@ -0,0 +1,85 @@
+package weather.geo
+
+import common.ResourcesHelper
+
+import scala.io.{BufferedSource, Source}
+import scala.util.Try
+
+case class CityRef(city: String, region: String, country: String)
+
+object CityRef {
+  def makeFixedCase(city: String, region: String, country: String): CityRef =
+    CityRef(city.toUpperCase(), region.toUpperCase(), country.toUpperCase())
+}
+
+object CitiesCsvLine {
+  implicit class RichString(s: String) {
+    def withoutQuotes: String = s.stripPrefix("\"").stripSuffix("\"")
+  }
+
+  def unapply(line: String): Option[CitiesCsvLine] = {
+    line.split(",", -1) match {
+      case Array(
+            geonamesId,
+            city,
+            region,
+            country,
+            latitude,
+            longitude,
+          ) =>
+        Try {
+          CitiesCsvLine(
+            geonamesId.toInt,
+            city.withoutQuotes,
+            region.withoutQuotes,
+            country.withoutQuotes,
+            latitude.toDouble,
+            longitude.toDouble,
+          )
+        }.toOption
+
+      case _ => None
+    }
+  }
+}
+
+case class CitiesCsvLine(
+    geonamesId: Int,
+    city: String,
+    region: String,
+    country: String,
+    latitude: Double,
+    longitude: Double,
+)
+
+object CitiesLookUp extends ResourcesHelper {
+  private[geo] def getGeoIPCityInputStream: BufferedSource = {
+    Source.fromInputStream(getInputStream("city_region_country_lat_long.csv").get, "iso-8859-1")
+  }
+
+  private[geo] def getCsvLines: Iterator[CitiesCsvLine] = {
+    val csv = getGeoIPCityInputStream
+    csv.getLines().drop(2) collect { case CitiesCsvLine(csvLine) => csvLine }
+  }
+
+  def loadCache(): Map[CityRef, LatitudeLongitude] = {
+    // first two lines are Copyright info and field names
```

Comment:
Is this comment meant to live before the `drop(2)`?

Reply:
Nice catch - it should also be `drop(1)` as we don't have copyright info in there.